Camera2Basic/app/build.gradle

```diff
@@ -14,29 +16,32 @@
  * limitations under the License.
  */
 
-apply plugin: 'com.android.application'
-apply plugin: 'kotlin-android'
-apply plugin: 'kotlin-kapt'
-apply plugin: "androidx.navigation.safeargs"
+plugins {
+    id 'com.android.application'
+    id 'kotlin-android'
+    id 'kotlin-kapt'
+    id 'androidx.navigation.safeargs.kotlin'
+}
 
 android {
-    compileSdkVersion 29
+    compileSdk 32
+
     defaultConfig {
-        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
         applicationId "com.android.example.camera2.basic"
-        minSdkVersion 21
-        targetSdkVersion 29
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+        minSdk 21
+        targetSdk 32
         versionCode 1
         versionName "1.0.0"
     }
 
     compileOptions {
-        sourceCompatibility rootProject.ext.java_version
-        targetCompatibility rootProject.ext.java_version
+        sourceCompatibility JavaVersion.VERSION_11
```

Comment:
Why hard code the Java version here? If you're trying to update from v8 to v11, can you just update it in one place in the root level build.gradle file? (Here and throughout the other examples)

Reply:
[]()**__![Screenshot_2022-06-21-05-02-05-94_6012fa4d4ddec268fc5c7112cbb265e7.jpg](https://user-images.githubusercontent.com/103907129/173865491-eb610c2c-3dae-4903-9a0a-46803975b06b.jpg)__**

Reply:
All the other modules have that hard-coded, too... and the reason is, that there is no more `buildscript.ext`. Would need to add the `buildscript` block back in, only to define `ext`. Just combed through once again and updated some more; all the latest/stable, while possible.